Source/Events.Handling/Builder/ConventionEventHandlerBuilder.cs

```diff
@@ -0,0 +1,311 @@
+// Copyright (c) Dolittle. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Dolittle.SDK.DependencyInversion;
+using Dolittle.SDK.Events.Handling.Internal;
+using Dolittle.SDK.Events.Processing;
+using Microsoft.Extensions.Logging;
+
+namespace Dolittle.SDK.Events.Handling.Builder
+{
+    /// <summary>
+    /// Methods for building <see cref="IEventHandler"/> instances by convention from an instantiated event handler class.
+    /// </summary>
+    public class ConventionEventHandlerBuilder : ICanBuildAndRegisterAnEventHandler
+    {
+        const string MethodName = "Handle";
+        readonly Type _eventHandlerType;
+        object _eventHandlerInstance;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConventionEventHandlerBuilder"/> class.
+        /// </summary>
+        /// <param name="eventHandlerInstance">The event handler instance.</param>
+        public ConventionEventHandlerBuilder(object eventHandlerInstance)
+        {
+            _eventHandlerInstance = eventHandlerInstance;
+            _eventHandlerType = eventHandlerInstance.GetType();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ConventionEventHandlerBuilder"/> class.
+        /// </summary>
+        /// <param name="eventHandlerType">The event handler <see cref="Type" />.</param>
+        public ConventionEventHandlerBuilder(Type eventHandlerType) => _eventHandlerType = eventHandlerType;
+
+        /// <inheritdoc/>
+        public BuildEventHandlerResult BuildAndRegister(
+            IEventProcessors eventProcessors,
+            IEventTypes eventTypes,
+            IEventProcessingConverter processingConverter,
+            IContainer container,
+            ILoggerFactory loggerFactory,
+            CancellationToken cancellation)
+        {
+            _eventHandlerInstance ??= container.Get(_eventHandlerType);
+            var logger = loggerFactory.CreateLogger<ConventionEventHandlerBuilder>();
+            logger.LogTrace("Building event handler from type {EventHandler}", _eventHandlerType);
```

Comment:
Seems like a debug message
```suggestion
            logger.LogDebug("Building event handler from type {EventHandler}", _eventHandlerType);
```